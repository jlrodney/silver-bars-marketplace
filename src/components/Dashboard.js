import React from 'react';
import Header from './Header';
import AddOrderForm from './AddOrderForm';
import Orders from './Orders';
import CombinedOrders from './CombinedOrders';

export default class Dashboard extends React.Component {
  state = {
    buyOrders:[],
    sellOrders:[],
  };

  addOrder = (order) => {
    if (order.type === 'BUY') {
      this.setState((prevState) => ({
        buyOrders: prevState.buyOrders.concat(order)
      }));
    } else if (order.type === "SELL") {
      this.setState((prevState) => ({
        sellOrders: prevState.sellOrders.concat(order)
      }));
    }
  };



  render() {
    return (
      <div>
        <Header />
        <p>Combined Sells</p>
        <CombinedOrders orders={this.state.sellOrders}/>
        <br/>
        <p>Combined Buys</p>
        <CombinedOrders orders={this.state.buyOrders}/>
        <AddOrderForm onSubmit={this.addOrder} />
      </div>
    );
  }
}
