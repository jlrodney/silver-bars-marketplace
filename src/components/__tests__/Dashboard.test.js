import React from 'react';
import { shallow, mount } from 'enzyme';
import Dashboard from '../Dashboard';

const buyOrder = {
    orderId: 'xyz986',
    userId: '1',
    quantity: 22,
    price: 230,
    type: 'BUY'
};

const sellOrder = {
    orderId: 'xyz998',
    userId: '1',
    quantity: 222,
    price: 230,
    type: 'SELL'
};

describe('Dashboard component', () => {
  it('should render Dashboard correctly', () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit method', () => {
  it('should update state of buyOrders accordingly', () => {
    const wrapper = mount(<Dashboard />);
    let value = 'user1';
    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    value = '33';
    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    value = 320;
    wrapper.find('input').at(2).simulate('change', {
      target: { value }
    });
    value = "BUY";
    wrapper.find('select').simulate('change', {
      target: { value }
    });
    wrapper.find('form').simulate('submit', {
      preventDefault: () => { }
    });
    expect(wrapper.state().buyOrders.length).toEqual(1)
  });

  it('should update state of sellOrders accordingly and then delete it', () => {
    const wrapper = mount(<Dashboard />);
    let value = 'user1';
    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    value = '33';
    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    value = 320;
    wrapper.find('input').at(2).simulate('change', {
      target: { value }
    });
    value = "SELL";
    wrapper.find('select').simulate('change', {
      target: { value }
    });
    wrapper.find('form').simulate('submit', {
      preventDefault: () => { }
    });
    expect(wrapper.state().sellOrders.length).toEqual(1);
    wrapper.find('.removeOrderButton').simulate('click');
    expect(wrapper.state().sellOrders.length).toEqual(0);
  });

  it('should update state of buy orders and then delet it', () => {
    const wrapper = mount(<Dashboard />);
    let value = 'user1';
    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    value = '33';
    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    value = 320;
    wrapper.find('input').at(2).simulate('change', {
      target: { value }
    });
    value = "BUY";
    wrapper.find('select').simulate('change', {
      target: { value }
    });
    wrapper.find('form').simulate('submit', {
      preventDefault: () => { }
    });
    expect(wrapper.state().buyOrders.length).toEqual(1);
    wrapper.find('.removeOrderButton').simulate('click');
    expect(wrapper.state().buyOrders.length).toEqual(0);
  });
});
