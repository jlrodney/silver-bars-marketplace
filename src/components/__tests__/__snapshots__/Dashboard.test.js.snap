// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should render Dashboard correctly 1`] = `
<div>
  <Header
    subtitle="Live order board"
    title="SilverBars Marketplace"
  />
  <p>
    Combined Sells
  </p>
  <CombinedOrders
    orders={Array []}
  />
  <br />
  <p>
    Combined Buys
  </p>
  <CombinedOrders
    orders={Array []}
  />
  <AddOrderForm
    onSubmit={[Function]}
  />
</div>
`;
